### PR TITLE
Allow search class using elastic search

### PR DESCRIPTION
**Pull request by gabrielfs7**
Allow search class using elastic search

Related PRs:

- [ ] https://github.com/oat-sa/lib-tao-elasticsearch/pull/43
- [ ] https://github.com/oat-sa/tao-core/pull/2790
- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/4

https://oat-sa.atlassian.net/browse/CON-497

### CODE REVIEW

File: models/classes/Lists/Business/Service/ClassMetadataSearcherProxy.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            $this->logCritical(
+                sprintf(
+                    'Impossible to perform class metadata search with %s: %s',
+                    $activeSearcherId,
+                    $exception->getMessage()
+                )
+            );
+
+            if ($activeSearcherId !== ClassMetadataService::SERVICE_ID) {
+                return $this->getClassMetadataSearcher()->findAll($input);
```
Comment (shpran) [inline models/classes/Lists/Business/Service/ClassMetadataSearcherProxy.php:58]: Should we wrap this into `try...catch` block as you did above?

**Review by shpran — APPROVED** [review]
LGTM 👍🏻 
Left one comment

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/Lists/Business/Service/ClassMetadataSearcherProxy.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/Lists/Business/Service/ClassMetadataSearcherProxy.php:58]: This was the previous behavior and already handled outside. The try / catch in this case is only for fallback cases

**Review by bartlomiejmarszal — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by bartlomiejmarszal — APPROVED** [review]
Tested on my local machine

**Pull request merged**